src/benchmarklib/tpcc/tpcc_table_generator.hpp

```diff
@@ -48,14 +50,17 @@ class TpccTableGenerator : public opossum::AbstractBenchmarkTableGenerator {
 
   std::shared_ptr<Table> generate_new_order_table();
 
-  std::map<std::string, std::shared_ptr<Table>> generate_all_tables();
+  std::map<std::string, std::shared_ptr<Table>> generate_all_tables() override;
 
-  static TpccTableGeneratorFunctions tpcc_table_generator_functions();
+  static TpccTableGeneratorFunctions table_generator_functions();
 
-  static std::shared_ptr<Table> generate_tpcc_table(const std::string& tablename);
+  std::shared_ptr<Table> generate_table(const std::string& table_name);
+
+  void encode_table(const std::string& table_name, std::shared_ptr<Table> table);
```

Comment:
Having this function in the public interface alongside "generate_table" is confusing to me. From the interface I'd assume that I need to call both of them if I want a table to be encoded - which is not the case from reading the code.

Imo, this should be private.

Reply:
true. will change.